Jenkinsfile

```diff
@@ -1,28 +1,69 @@
 pipeline {
     agent any
 
+    tools { 
+        maven 'maven_3_5_4' 
+        jdk 'jdk8' 
+    }
+    environment {
+        // requires the Pipeline Utility Steps plugin
+        pom = readMavenPom file: 'pom.xml'
+        pomVersion = pom.getVersion()
+        pomArtifactId = pom.getArtifactId()
+        //pomReleases = pom.getProperties().get("artifactory.releases")
+        //pomSnapshots = pom.getProperties().get("artifactory.snapshots")
+    }
+
     stages {
-        stage('Package') {
+        stage('Rest Repo') {
+            // it is not clear that the repo is a clean checkout
             steps {
-                withMaven(maven:'maven_3_5_4') {
-                    sh 'mvn clean package'
-                }
+                // remove potentially old release files if exist
+                sh 'rm pom.xml.releaseBackup release.properties 2>/dev/null || true'
```

Comment:
I normally clean the workspace and use the scm plugin to do a clean pull, e.g.
```  stages {
    stage('Clean Workspace') {
      steps {
        cleanWs()
      }
    }
    stage('git clone') {
      steps{
        checkout(
          [
            $class: 'GitSCM',
            branches: [[name: "${GIT_BRANCH}"]],
            doGenerateSubmoduleConfigurations: false,
            extensions: [
              [
                $class: 'LocalBranch',
                localBranch: "${GIT_BRANCH}"
              ]
            ],
            submoduleCfg: [],
            userRemoteConfigs: [
              [
                credentialsId: 'CIDA-Jenkins-GitHub',
                url: 'https://github.com/${REPO_PATH}.git'
              ]
            ]
          ]
        )
      }
    }
```

Reply:
That may or may not be useful to you...

Reply:
Reading this, it looks like the clean and checkout is more complicated since the pipeline already checked it out. I heard you say that you have stored the Jenkinsfile in a different repo.